src/main/java/org/mozilla/iot/webthing/Property.java

```diff
@@ -93,18 +90,19 @@ private void validateValue(T value) throws PropertyError {
      * @return Description of the property as an object.
      */
     public JSONObject asPropertyDescription() {
-        List<String> keyList = new ArrayList<>();
-        this.metadata.keys().forEachRemaining((key) -> {
-            keyList.add(key);
-        });
-        String[] keys = keyList.toArray(new String[keyList.size()]);
-        JSONObject description = new JSONObject(this.metadata, keys);
-        JSONArray links = new JSONArray();
+        JSONObject description = new JSONObject(this.metadata.toString());
         JSONObject link = new JSONObject();
         link.put("rel", "property");
         link.put("href", this.hrefPrefix + this.href);
-        links.put(link);
-        description.put("links", links);
+
+        if (description.has("links")) {
+            description.getJSONArray("links").put(link);
```

Comment:
I would not have expected `getJSONArray` to return a mutable reference to the actual underlying object but I guess that's just Java for you